### PR TITLE
Add exact filters for relationships

### PR DESCRIPTION
**Pull request by caffeineinc**
Check to see if the field referenced is a relationship on the instance, if it's found, then we should do an exact match over a partial match, so part ID's aren't returned. 

@see issue #49

(WIP as i need to do tests still)

### CODE REVIEW

**Comment by caffeineinc** [comment]
Ok, so this should cover filtering by a relationship.

**Comment by robbieaverill** [comment]
The tests are failing

**Comment by caffeineinc** [comment]
Yeah, I believe these are environmental as the same tests don't fail in my local. I'll just try running them again and see if investigate why.

**Comment by ScopeyNZ** [comment]
Try reproducing the exact steps taken by travis.

- `git clone silverstripe/silverstripe-registry` (hub allows you to omit "github.com")
- `composer require --no-update silverstripe/installer 4.0.x-dev`
- `composer install`
- `vendor/bin/phpunit`

**Comment by NightJar** [comment]
>Even @NightJar confirmed it passes, so i'm not going insane.

Yes.

```
[vagrant@archlinux ~]$ php -v
PHP 7.1.26 (cli) (built: Mar  7 2019 02:42:14) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
[vagrant@archlinux ~]$ mysql --version
mysql  Ver 15.1 Distrib 10.3.13-MariaDB, for Linux (x86_64) using readline 5.1
[vagrant@archlinux ~]$
```

```
[vagrant@archlinux silverstripe-registry]$ history | tail
  503  git clone git@github.com:caffineinc/silverstripe-registry.git
  504  git clone git@github.com:caffeineinc/silverstripe-registry.git
  505  cd silverstripe-registry/
  506  cat .travis.yml
  507  composer validate
  508  composer require --no-update silverstripe/installer 4.0.x-dev
  509  composer install --prefer-dist --no-interaction --no-progress --no-suggest --optimize-autoloader --verbose --profile
  510  cat .travis.yml
  511  vendor/bin/phpunit
  512  history | tail
[vagrant@archlinux silverstripe-registry]$
```

```
[vagrant@archlinux silverstripe-registry]$ vendor/bin/phpunit
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

.....................                                             21 / 21 (100%)

Time: 6.17 minutes, Memory: 78.00MB

OK (21 tests, 45 assertions)
[vagrant@archlinux silverstripe-registry]$
```

**Comment by NightJar** [comment]
>Try reproducing the exact steps taken by travis.

I did exactly this, as I suspected the version lock to `4.0.x-dev` to be issue causing; but that hypothesis was disproven.

**Comment by NightJar** [comment]
hurr durr history shows I cloned correctly... but didn't check out :facepalm:

re-retesting

File: tests/RegistryPageFunctionalTest.php [hunk 1]
```diff
@@ -82,6 +82,37 @@ public function testFilteredByRelationSearchResults()
         $this->assertEquals('Sherson', (string) $cells[1]->a[0]);
     }
 
+    /**
+     * Check that RegistryPageController can filter for ExactMatches (ID) for relationships.
+     *
+     * @throws \Exception
```
Comment (robbieaverill) [inline tests/RegistryPageFunctionalTest.php]: Tests shouldn't throw exceptions. Are you expecting one here instead?

**Comment by caffeineinc** [comment]
HAHAHA FINALLY `All checks have passed` 

@robbieaverill @ScopeyNZ @NightJar

**Comment by caffeineinc** [comment]
They were somewhat legit bugs in the tests, my change added a new column, and i made the mistake of assuming that the ID of the fixture was always 2.  (it apparently differs on PHP 7.1).

File: tests/RegistryPageFunctionalTest.php (hunk 1 above)
Comment (caffeineinc) [reply tests/RegistryPageFunctionalTest.php]: tests do throw exceptions. $this->assert does exactly that doesn't it. I think it was PHP storm adding things for me. I can remove it if it makes you happier.

Comment (robbieaverill) [reply tests/RegistryPageFunctionalTest.php]: Yeah if you don't mind, we don't usually annotate phpunit's internal exceptions

**Pull request merged**